### PR TITLE
Mttl 1760

### PR DESCRIPTION
**Pull request by evak6979**
## Link to JIRA ticket

[Add a link to the JIRA ticket that the changes in this PR describe.](https://hackney.atlassian.net/browse/MTTL-1760)

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

Are there any next steps that need to addressed after merging this PR? Add them here.

### CODE REVIEW

File: EqualityInformationApi/V1/Controllers/GetEqualityInformationApiController.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        }
+
+        [ProducesResponseType(typeof(List<EqualityInformationObject>), StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [HttpGet]
+        [LogCall(LogLevel.Information)]
+        public async Task<IActionResult> Get([FromQuery] Guid targetId)
+        {
+            if (targetId == Guid.Empty)
```
Comment (amidomichael) [inline EqualityInformationApi/V1/Controllers/GetEqualityInformationApiController.cs:43]: should this not be a validator?

File: EqualityInformationApi/V1/Gateways/EqualityInformationGateway.cs [hunk 2]
```diff
@@ -47,5 +46,17 @@ public async Task<EqualityInformation> Update(PatchEqualityInformationObject req
...
         }
+
+        [LogCall]
+        public async Task<EqualityInformation> Get(Guid targetId)
+        {
+            var entity = await _dynamoDbContext.QueryAsync<EqualityInformationDb>(targetId).GetNextSetAsync()
+                .ConfigureAwait(false);
+
+            if (entity == null || !entity.Any())
+                return null;
```
Comment (amidomichael) [inline EqualityInformationApi/V1/Gateways/EqualityInformationGateway.cs:57]: shouldn't this throw an exception?

File: EqualityInformationApi/V1/UseCase/GetUseCase.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            ISnsFactory snsFactory)
+        {
+            _gateway = gateway;
+        }
+
+        public async Task<EqualityInformationResponseObject> Execute(Guid targetId, Token token)
+        {
+            var equalityInformation = await _gateway.Get(targetId).ConfigureAwait(false);
+
+            return equalityInformation?.ToResponse();
```
Comment (amidomichael) [inline EqualityInformationApi/V1/UseCase/GetUseCase.cs:28]: shouldn't it return a 404 if not found?

File: EqualityInformationApi/V1/Controllers/GetEqualityInformationApiController.cs (hunk 1 above)
Comment (evak6979) [reply EqualityInformationApi/V1/Controllers/GetEqualityInformationApiController.cs:43]: the validators require an object type of T for fluent validation (From what I saw - e.g. AbstractValidator<PatchEqualityInformationObject> ). This is just an incoming string request.

File: EqualityInformationApi/V1/Gateways/EqualityInformationGateway.cs (hunk 2 above)
Comment (evak6979) [reply EqualityInformationApi/V1/Gateways/EqualityInformationGateway.cs:57]: If not found, then it should be not found - it's treated in a controller level. Would you rather the controller catches an exception and returns not found?

File: EqualityInformationApi/V1/UseCase/GetUseCase.cs (hunk 3 above)
Comment (evak6979) [reply EqualityInformationApi/V1/UseCase/GetUseCase.cs:28]: It does on controller level.

File: EqualityInformationApi/V1/Controllers/GetEqualityInformationApiController.cs (hunk 1 above)
Comment (amidomichael) [reply EqualityInformationApi/V1/Controllers/GetEqualityInformationApiController.cs:43]: isn't `string` a type?

File: EqualityInformationApi/V1/Gateways/EqualityInformationGateway.cs (hunk 2 above)
Comment (amidomichael) [reply EqualityInformationApi/V1/Gateways/EqualityInformationGateway.cs:57]: yep :-)

File: EqualityInformationApi/V1/UseCase/GetUseCase.cs (hunk 3 above)
Comment (amidomichael) [reply EqualityInformationApi/V1/UseCase/GetUseCase.cs:28]: ignore me, was being dumb

File: EqualityInformationApi/V1/Controllers/GetEqualityInformationApiController.cs (hunk 1 above)
Comment (evak6979) [reply EqualityInformationApi/V1/Controllers/GetEqualityInformationApiController.cs:43]: I can try and see if I can make it work.

**Review by amidomichael — APPROVED** [review]

**Pull request merged**